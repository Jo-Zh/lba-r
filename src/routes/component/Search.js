import React from "react";

const Search = () => {
  return (
    <form role="search" method="POST" className="form-group bg-light">
      <div className="input-group input-group-sm">
        <input
          type="text"
          name="search"
          className="text-dark my-1"
          placeholder="Search for "
          aria-describedby="inputGroup-sizing-sm"
        />
        <button type="submit" className="btn btn-outline-warning btn-sm my-1">
          <i className="bi bi-search">IN</i>
        </button>
        <select
          className="rounded my-1 "
          name="field"
          id="floatingSelectGrid"
          defaultValue={"content"}
        >
          <option value="title">title</option>
          <option value="content">content</option>
          <option value="writer">poster</option>
        </select>
      </div>
    </form>
  );
};
export default Search;
