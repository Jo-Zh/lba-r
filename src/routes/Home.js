import React from "react";
import Article from "./component/Article";
import Search from "./component/Search";
import Category from "./component/Category";

const Home = (props) => {
  return (
    <div>
      <div className="row text-center py-3">
        <h1>Explore and Sharing learning experience here... </h1>
      </div>
      <div className="d-flex flex-spacebetween bg-light">
        <Search />
        <Category />
      </div>

      <div className="container-fluid px-4 my-4">
        <div className="row g-4 ">
          {props.posts.map((post) => {
            return (
              <Article
                key={post.id}
                postcategory={post.category}
                postimage={post.cover}
                posttitle={post.title}
                postcreater={post.creater}
                postdate={post.date}
                postcontent={post.content}
                id={post.id}
              />
            );
          })}
          <div className="col">No results</div>
        </div>
      </div>
    </div>
  );
};

export default Home;
