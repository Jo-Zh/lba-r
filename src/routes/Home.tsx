import React from "react";
import Article from "./component/Article";
import Search from "./component/Search";
import Category from "./component/Category";

const Home = ({ posts }: HomeProps) => {
  return (
    <div>
      <div className="row text-center py-3">
        <h1>Explore and Sharing learning experience here... </h1>
      </div>
      <div className="d-flex flex-spacebetween bg-light">
        <Search />
        <Category />
      </div>

      <div className="container-fluid px-4 my-4">
        <div className="row g-4 ">
          {posts.map((post) => {
            return (
              <Article
                key={post.id}
                postcategory={post.category}
                postimage={post.cover}
                posttitle={post.title}
                postcreater={post.creater}
                postdate={post.date}
                postcontent={post.content}
                id={post.id}
              />
            );
          })}
          <div className="col">No results</div>
        </div>
      </div>
    </div>
  );
};

type HomeProps = {
  posts: {
    id: number;
    title: string;
    content: string;
    category: string;
    creater: string;
    date: Date;
    cover?: string;
  }[];
};

export default Home;
