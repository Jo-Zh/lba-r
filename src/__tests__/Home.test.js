import Search from "../routes/component/Search";
import React from "react";
import Home from "../routes/Home";
import { BrowserRouter as Router } from "react-router-dom";
import TestRenderer from "react-test-renderer";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders correctly", () => {
  const tree = TestRenderer.create(<Search />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders with or without a name", () => {
  act(() => {
    const posts = [
      {
        id: 1,
        title: "example",
        content: "something here",
        category: "code-learning",
        cover: null,
      },
      {
        id: 2,
        title: "example",
        content: "something here",
        category: "code-learning",
        cover: null,
      },
    ];

    render(
      <Router>
        <Home posts={posts} />
      </Router>,
      container
    );
  });

  expect(container.querySelectorAll("#test-article").length).toEqual(2);
});
