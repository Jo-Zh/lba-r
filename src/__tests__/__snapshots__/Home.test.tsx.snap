// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="form-group bg-light"
  method="POST"
  role="search"
>
  <div
    className="input-group input-group-sm"
  >
    <input
      aria-describedby="inputGroup-sizing-sm"
      className="text-dark my-1"
      name="search"
      placeholder="Search for "
      type="text"
    />
    <button
      className="btn btn-outline-warning btn-sm my-1"
      type="submit"
    >
      <i
        className="bi bi-search"
      >
        IN
      </i>
    </button>
    <select
      className="rounded my-1 "
      defaultValue="content"
      id="floatingSelectGrid"
      name="field"
    >
      <option
        value="title"
      >
        title
      </option>
      <option
        value="content"
      >
        content
      </option>
      <option
        value="writer"
      >
        poster
      </option>
    </select>
  </div>
</form>
`;
